packages/wonder-blocks-button/components/button-core.js

```diff
@@ -0,0 +1,271 @@
+// @flow
+import React from "react";
+import {StyleSheet, css} from "aphrodite";
+
+import {LabelLarge} from "wonder-blocks-typography";
+import {processStyleList} from "../../wonder-blocks-core/util/util.js";
+import Color, {mix, fade} from "wonder-blocks-color";
+import type {ValidTints} from "wonder-blocks-color";
+import type {SharedProps} from "../types.js";
+
+type Props = SharedProps & {
+    hovered?: boolean,
```

Comment:
I think it might make sense to have all three of these be required. ButtonCore is only used internally and we want to make sure that we never forget to pass in one of these options (that would be a bad sign if we forgot!)

Reply:
Another option is to have a single required prop called `state`, e.g.
```
state: "default" | "hovered" | "pressed" | "focused" | "disabled"
```
That we we don't have to pass `focused={false} pressed={false} hovered={true}` or combinations thereof.

Reply:
My understanding is that a `Button` can be in combinations of those states? Or do you mean since `ButtonCore` only displays any one of those states at a time, we just pass in the state with most precedence in `ButtonBehavior`?